src/jira/client/jira-api-client.ts

```diff
@@ -0,0 +1,494 @@
+	JiraDeploymentBulkSubmitData,
+	JiraIssue,
+	JiraSubmitOptions,
+	JiraVulnerabilityBulkSubmitData
+} from "interfaces/jira";
+
+const issueKeyLimitWarning = "Exceeded issue key reference limit. Some issues may not be linked.";
+
+export interface DeploymentsResult {
+	status: number;
+	rejectedDeployments?: any[];
+}
+
+export class JiraClient {
+	axios: AxiosInstance;
+	logger: Logger;
+	jiraHost: string;
+	gitHubInstallationId: number;
+	gitHubAppId: number | undefined;
+
+	static async create(installation: Installation, gitHubAppId: number | undefined, logger: Logger): Promise<JiraClient> {
+		const jiraClient = new JiraClient(installation.jiraHost, installation.id, gitHubAppId, logger);
+		await jiraClient.initialize(installation);
+		return jiraClient;
+	}
+
+	private async initialize(installation: Installation): Promise<void> {
+		const secret = await installation.decrypt("encryptedSharedSecret", this.logger);
+		this.axios = getAxiosInstance(installation.jiraHost, secret, this.logger);
+	}
+
+	private constructor(jiraHost: string, gitHubInstallationId: number, gitHubAppId: number | undefined, logger: Logger) {
+		const gitHubProduct = getCloudOrServerFromGitHubAppId(gitHubAppId);
+
+		this.jiraHost = jiraHost;
+		this.gitHubInstallationId = gitHubInstallationId;
+		this.gitHubAppId = gitHubAppId;
+		this.logger = logger.child({ jiraHost, gitHubInstallationId, gitHubAppId, gitHubProduct });
+	}
+
+	async isAuthorized(): Promise<boolean> {
+		try {
+			return (await this.axios.get("/rest/devinfo/0.10/existsByProperties?fakeProperty=1")).status === 200;
+		} catch (error) {
+			if (!(error instanceof JiraClientError)) {
+				throw error;
+			}
+			return false;
+		}
+	}
+
+	async getCloudId(): Promise<{ cloudId: string }> {
+		return (await this.axios.get("_edge/tenant_info")).data;
+	}
+
+	async appPropertiesCreate(isConfiguredState: boolean) {
+		return await this.axios.put(`/rest/atlassian-connect/latest/addons/${envVars.APP_KEY}/properties/is-configured`, {
+			"isConfigured": isConfiguredState
+		});
+	}
+
+	async appPropertiesGet() {
+		return await this.axios.get(`/rest/atlassian-connect/latest/addons/${envVars.APP_KEY}/properties/is-configured`);
+	}
+
+	async appPropertiesDelete() {
+		return await this.axios.delete(`/rest/atlassian-connect/latest/addons/${envVars.APP_KEY}/properties/is-configured`);
+	}
+
+	async linkedWorkspace(subscriptionId: number) {
+		const payload = {
+			"workspaceIds": [subscriptionId]
+		};
+		return await this.axios.post("/rest/security/1.0/linkedWorkspaces/bulk", payload);
+	}
+
+	async deleteWorkspace(subscriptionId: number) {
+		return await this.axios.delete(`/rest/security/1.0/linkedWorkspaces/bulk?workspaceIds=${subscriptionId}`);
+	}
+
+	// TODO TEST
+	async checkAdminPermissions(accountId: string) {
+		const payload = {
+			accountId,
+			globalPermissions: [
+				"ADMINISTER"
+			]
+		};
+		return await this.axios.post("/rest/api/latest/permissions/check", payload);
+	}
+
+	// TODO TEST
```

Comment:
Lots of TODO TEST comments, left there intentionally, after this PR I will add a test block for each of these functions....